src/comm_player_manager.c

```diff
@@ -388,48 +388,46 @@ static void CommPlayer_Add (u8 netId)
         return;
     }
 
-    {
-        TrainerInfo * trainerInfo = CommInfo_TrainerInfo(netId);
+    TrainerInfo * trainerInfo = CommInfo_TrainerInfo(netId);
 
-        if (trainerInfo) {
-            if (!sCommPlayerManager->isUnderground) {
-                if (netId != CommSys_CurNetId()) {
-                    MapObject * obj = MapObjMan_LocalMapObjByIndex(sCommPlayerManager->fieldSys->mapObjMan, 0xff + netId + 1);
+    if (trainerInfo) {
+        if (!sCommPlayerManager->isUnderground) {
+            if (netId != CommSys_CurNetId()) {
+                MapObject * obj = MapObjMan_LocalMapObjByIndex(sCommPlayerManager->fieldSystem->mapObjMan, 0xff + netId + 1);
 
-                    if (obj != 0) {
-                        MapObject_Delete(obj);
-                    }
+                if (obj != NULL) {
+                    MapObject_Delete(obj);
                 }
             }
+        }
```

Comment:
Might be more readable to condense these `if` chains to a single compound:

```c
if (trainerInfo
        && !sCommPlayerManager->isUnderground
        && netId != CommSys_CurNetId()) {
    MapObject * obj = MapObjMan_LocalMapObjByIndex(sCommPlayerManager->fieldSystem->mapObjMan, 0xff + netId + 1);
    
    if (obj != NULL) {
        MapObject_Delete(obj);
    }
}
```

Reply:
that obviously doen't match though? the second two can be condensed but ``trainerInfo`` has more in its block

Reply:
Excess scope blocks got me confused... at least merge the ones that can be merged. Fewer indentation levels are good.